### DISCUSSION TITLE
Add component "Deconcatenation"

### DISCUSSION THREAD

**Discussion by MaaiG** [FakeFishGames/Barotrauma#7648] [Ideas and suggestions]
~~Add a component `Deconcatenation` to the reverse component `Concatenation`.
This will allow multiple signals to be sent over one channel. Or take the sample from the one long signal.
For example: I want to send a complex composite signal to a wireless component and the receiver has to separate it.
I want to read the remote drone telemetry by the one channel or something like that etc.~~
There was no need! Use RegEx!
P.S. I wanted to have one component instead of several RegEx, but it’s okay.

**Comment by mLuby** [comment]
Sounds like you want the [Regex Component](https://barotraumagame.com/wiki/Wiring_Components#RegEx_Find_Component). <img alt="regex sprite" width="30" src="https://barotraumagame.com/baro-wiki/images/7/7d/RegEx_Component.png" />

For example imagine you concat(concat(`velocity_x` `","`) `velocity_y`). You can extract the values by passing that signal to two Regex components: regex

```
NavTerminal.velocity_x -> in_1.ConcA
Mem{value=,} -> in_2.ConcA.out -> in_1.ConcB.out -> RegexA{output:vel_x, expression:(?<vel_x>[0-9.]+),} -> DisplayA
NavTerminal.velocity_y -> in_2.ConcB.out ->RegexB{output:vel_y, expression:,(?<vel_y>[0-9.]+)} -> DisplayB
```

I'd suggest visiting the [Barotrauma Official Discord](https://discordapp.com/invite/undertow) to ask around in #baro-sub-building and #baro-critique-and-ideas. Lots of helpful people there.

Other types of [multiplexing](https://en.wikipedia.org/wiki/Multiplexing) and how to do them in Barotrauma:
1. [space](https://en.wikipedia.org/wiki/Multiplexing#Space-division_multiplexing): use a separate wire for each signal.
2. [frequency](https://en.wikipedia.org/wiki/Multiplexing#Frequency-division_multiplexing): use a separate wifi channel for each signal.
3. [time](https://en.wikipedia.org/wiki/Multiplexing#Time-division_multiplexing): use one wire/channel and agree to send/receive a given signal at distinct time intervals. Example:
```
Oscillator{type=sawtooth} -> Regex{output:1, false:0, expression:0.3} -> set_state.Relay.signal_1_in/out <-> signal
```
![](https://upload.wikimedia.org/wikipedia/commons/e/e0/Telephony_multiplexer_system.gif)

**Reply by MaaiG** [reply]
> Sounds like you want the [Regex Component](https://barotraumagame.com/wiki/Wiring_Components#RegEx_Find_Component).

> Thanks! I’ll certainly think about it. )

I thought about it.
I’ve always lazy looked at regular expressions... not my subject, but now... Thanks for this "kick in the ass". :)
I got more involved in the regular expressions and RegEx component. 😎
I realized that idea! It works.
- `\d+(?=\|)`
- `(?<=\|)\d+`
- It's alive! ))
![Image 003_23-15](https://user-images.githubusercontent.com/93348499/147838208-cb106d42-635a-47dd-97a6-5b64b18f0fda.png)
